### PR TITLE
[Tunnel PFC][Fix bug] Fix bug and Tests for adding property 'sai_remap_prio_on_tnl_egress'

### PR DESCRIPTION
**Pull request by richardyu-ms**
#### Why I did it
    Fix a bug in the template
    Add tests for adding property 'sai_remap_prio_on_tnl_egress', this
    property should only be added in dual tor environment.
    

#### How I did it
Add tests for adding property 'sai_remap_prio_on_tnl_egress'
#### How to verify it
    Test done:
    Run test test_j2files.py
```
python3 -m pytest src/sonic-config-engine/tests/test_j2files.py
=================================================================== test session starts ===================================================================
platform linux -- Python 3.7.3, pytest-3.10.1, py-1.7.0, pluggy-0.8.0
rootdir: /data/forks/sonic-buildimage/src/sonic-config-engine, inifile:
plugins: cov-2.6.0
collected 27 items                                                                                                                                        

src/sonic-config-engine/tests/test_j2files.py ...........................                                                                           [100%]

=============================================================== 27 passed in 18.84 seconds ================================================================

```
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Comment by richardyu-ms** [comment]
this pr will be merged after https://github.com/Azure/sonic-buildimage/pull/11027

File: src/sonic-config-engine/tests/test_j2files.py [hunk 1]
```diff
@@ -323,6 +323,42 @@ def test_qos_dscp_remapping_render_template(self):
             assert utils.cmp(sample_output_file, test_output)
             os.remove(test_output)
 
+    def test_config_brcm_render_template(self):
+        if utils.PYvX_DIR != 'py3':
+            #Skip on python2 as the change will not be backported to previous version
+            return
+
+        dir_paths = [
+            '../../../device/arista/x86_64-arista_7050cx3_32s/Arista-7050CX3-32S-D48C8',
```
Comment (lguohan) [inline src/sonic-config-engine/tests/test_j2files.py:332]: i suggest to get the config.bcm local to the tests folder, to make the package self-contained.

**Comment by lguohan** [comment]
please rebase the pr and test.

File: src/sonic-config-engine/tests/test_j2files.py (hunk 1 above)
Comment (richardyu-ms) [reply src/sonic-config-engine/tests/test_j2files.py:348]: yes, the j2 generated file should be there, added.

File: src/sonic-config-engine/tests/sample_output/py3/arista7260-dualtor.config.bcm [hunk 2]
```diff
@@ -0,0 +1,943 @@
+-%}l3_alpm_hit_skip=1
```
Comment (neethajohn) [inline src/sonic-config-engine/tests/sample_output/py3/arista7260-dualtor.config.bcm:1]: remove the -%}. How are the tests passing with this?

Comment (richardyu-ms) [reply src/sonic-config-engine/tests/sample_output/py3/arista7260-dualtor.config.bcm:1]: there is a bug, fixed it.

File: src/sonic-config-engine/tests/test_j2files.py (hunk 1 above)
Comment (richardyu-ms) [reply src/sonic-config-engine/tests/test_j2files.py:348]: if we place a replica of the j2 template into test folder, then we need to keep them sync manully when they changed?

File: src/sonic-config-engine/tests/test_j2files.py [hunk 3]
```diff
@@ -372,7 +367,7 @@ def test_config_brcm_render_template(self):
           
             #check output config.bcm
             config_sample_output_file = os.path.join(self.test_dir, 'sample_output', utils.PYvX_DIR, config_sample_output)
-            assert utils.cmp(config_sample_output_file, config_test_output)
+            assert utils.cmp(config_sample_output_file, config_test_output)
```
Comment (neethajohn) [inline src/sonic-config-engine/tests/test_j2files.py:370]: trailing whitespaces?

Comment (richardyu-ms) [reply src/sonic-config-engine/tests/test_j2files.py]: fixed

**Review by neethajohn — APPROVED** [review]

**Review by lguohan — APPROVED** [review]

**Comment by richardyu-ms** [comment]
fix in 202012 merged in https://github.com/Azure/sonic-buildimage/commit/b0c48f9b31ccba088dbe9cd0981b8949d7ed6e47

**Pull request merged**